### PR TITLE
Update new-azurermvm simple param set to enable auto configuration of Accelerated Network for qualifying vms

### PR DESCRIPTION
**Pull request by praries880**
Fix for issue #6673

### CODE REVIEW

**Comment by praries880** [comment]
On Demand : https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/1005/

File: src/ResourceManager/Compute/Commands.Compute/Common/Utils.cs [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+            public ImageTypeForAccelNet(string publisher, string offer, string skuPattern)
+            {
+                Publisher = publisher;
+                Offer = offer;
+                SkuPattern = skuPattern;
+            }
+
+            public bool Matches(string publisher, string offer, string sku)
+            {
+                return (Publisher.Equals(publisher) && (offer == null || offer.Equals(Offer)) && (sku == null || SkuPattern == null || (new Regex(SkuPattern)).IsMatch(sku)));
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute/Common/Utils.cs]: @praries880 do we need to do a null-check on `publisher` and/or `Publisher` here? It seems we do so for the other parameters

File: src/ResourceManager/Compute/Commands.Compute/Common/Utils.cs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+            var sizeFound = ValidSizesForAccelNet.Where(x => x.ToLower().Equals(size.ToLower()));
+            if (sizeFound == null || sizeFound.Count() <= 0)
+            {
+                return false;
+            }
+
+            var fourCoreSize = FourCoreSizesForAccelNet.Where(x => x.ToLower().Equals(size.ToLower()));
+
+            if (fourCoreSize == null | fourCoreSize.Count() == 0)
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute/Common/Utils.cs]: @praries880 should the operator here be `||` ?

File: src/ResourceManager/Compute/Commands.Compute/Common/Utils.cs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            {
+                var locationToUse = location;
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute/Common/Utils.cs]: @praries880 nit: we can consolidate the next few lines into one:

```cs
var locationToUse = String.IsNullOrWhitespace(location) ? defaultLocation : location;
```

File: src/ResourceManager/Compute/Commands.Compute/Common/Utils.cs (hunk 1 above)
Comment (praries880) [reply src/ResourceManager/Compute/Commands.Compute/Common/Utils.cs]: We don't need to, we validate that the publisher is not null at the top of the function.

File: src/ResourceManager/Compute/Commands.Compute/Common/Utils.cs (hunk 2 above)
Comment (praries880) [reply src/ResourceManager/Compute/Commands.Compute/Common/Utils.cs]: :) why.. yes.. yes it does

**Comment by praries880** [comment]
@cormacpayne  took care of review comment

File: src/ResourceManager/Compute/Commands.Compute/Common/Utils.cs (hunk 1 above)
Comment (sergey-shandar) [reply src/ResourceManager/Compute/Commands.Compute/Common/Utils.cs]: @cormacpayne @praries880 we need Nullable Reference Types (from C# 8.0) https://blogs.msdn.microsoft.com/dotnet/2017/11/15/nullable-reference-types-in-csharp/

**Pull request merged**